src/routes/things.js

```diff
@@ -47,8 +57,100 @@ let router = express.Router();
  * }
  **/
 
-router.get('/Things', (req, res) => {
-  res.status(200).send();
+const resource = 'Things';
+
+router.get('/', (req, res) => {
+  db().then(models => {
+    if (req.params && req.params[0]) {
+      models.Things.findById(req.params[0], {
+        attributes: {
+          exclude: excludedFields
+        }
+      }).then(thing => {
+        if (!thing) {
+          return ApiError(res, 404, ERRNO_RESOURCE_NOT_FOUND, NOT_FOUND);
+        }
+        res.status(200).json(response.generate(thing));
+      }).catch(() => {
+        ApiError(res, 404, ERRNO_RESOURCE_NOT_FOUND, NOT_FOUND);
+      });
+    } else {
+      models.Things.findAll({
+        attributes: {
+          exclude: excludedFields
+        }
+      }).then(things => {
+        res.status(200).json(response.generate(things));
+      });
+    }
+  }).catch(() => {
+    ApiError(res, 500, ERRNO_INTERNAL_ERROR, INTERNAL_ERROR);
+  });
+});
+
+// XXX #9 [Things router] Implement associations to other models and integrity
+
+router.post('/', (req, res) => {
+  db().then(models => {
+    models.Things.create(req.body)
+    .then(createdThing => {
+      // XXX #13 [Things router] response urls should be absolute
+      res.location('/' + resource + '/' + createdThing.id);
+      res.status(201).send();
+    }).catch(() => {
+      res.status(400).send();
```

Comment:
You need to return an `ApiError` here. Like in https://github.com/mozilla-sensorweb/sensorthings/pull/7/files#diff-628bdb43f99bc87ce4d142a35da514d4R88